sdk/provisioning/Azure.Provisioning/src/BicepValueOfT.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using Azure.Provisioning.Expressions;
+using Azure.Provisioning.Primitives;
+
+namespace Azure.Provisioning;
+
+// TODO: See the comment on BicepValue about how this will be reshaped into
+// more strongly typed models.
+
+/// <summary>
+/// Represents a value or expression.
+/// </summary>
+/// <typeparam name="T">The type of the value.</typeparam>
+public class BicepValue<T> : BicepValue
+{
```

Comment:
`BicepValue<T>` is the workhorse.  It's either a `T` or an expression.  There are implicit casts so you can just say `acct.Sku = SkuName.H100` without explicitly constructing an instance.  This has some tradeoffs with intellisense, but I'm planning some changes that will minimize the displeasure.